test/conformance/generatename_test.go

```diff
@@ -0,0 +1,187 @@
+// +build e2e
+
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conformance
+
+import (
+	pkgTest "github.com/knative/pkg/test"
+	"github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	rtesting "github.com/knative/serving/pkg/reconciler/v1alpha1/testing"
+	"github.com/knative/serving/test"
+	"net/http"
+	"strings"
+	"testing"
+)
+
+func setServiceGenerateName(generateName string) rtesting.ServiceOption {
+	return func(service *v1alpha1.Service) {
+		service.ObjectMeta.GenerateName = generateName
+	}
+}
+
+func setConfigurationGenerateName(generateName string) rtesting.ConfigOption {
+	return func(config *v1alpha1.Configuration) {
+		config.ObjectMeta.GenerateName = generateName
+	}
+}
+
+func setRouteGenerateName(generateName string) rtesting.RouteOption {
+	return func(route *v1alpha1.Route) {
+		route.ObjectMeta.GenerateName = generateName
+	}
+}
+
+// getGenerateNamePrefix returns the object name to be used for testing, shorted to
+// 44 characters to avoid #3236, as generateNames longer than 44 characters may cause
+// some knative resources to never become ready.
+func getGenerateNamePrefix(t *testing.T) string {
+	generateName := test.ObjectNameForTest(t) + "-"
+
+	generateNameLength := len(generateName)
+	if generateNameLength > 44 {
+		generateNameLength = 44
+	}
+	return generateName[0:generateNameLength]
+}
+
+func ensureCanServeRequests(t *testing.T, clients *test.Clients, route *v1alpha1.Route) {
+	t.Logf("Route %s has a domain set in its status", route.Name)
+	var domain string
+	err := test.WaitForRouteState(
+		clients.ServingClient,
+		route.Name,
+		func(r *v1alpha1.Route) (bool, error) {
+			domain = r.Status.Domain
+			return domain != "", nil
+		},
+		"RouteDomain",
+	)
+
+	t.Logf("Route %s can serve the expected data at the endpoint", route.Name)
+	_, err = pkgTest.WaitForEndpointState(
+		clients.KubeClient,
+		t.Logf,
+		domain,
+		pkgTest.Retrying(pkgTest.EventuallyMatchesBody(helloWorldText), http.StatusNotFound),
```

Comment:
This was changed elsewhere to avoid masking 503s. Consider changing. 

https://github.com/knative/serving/commit/b782c6bf8028d069640e37edc9599b8c8b3c9bd9

Reply:
Changed to match the style of the commit you linked.